src/Emogrifier.php

```diff
@@ -1267,24 +1282,70 @@ private function splitCssAndMediaQuery($css)
-        $cssForAllowedMediaTypes = preg_replace_callback(
-            '#@media\\s+(?:only\\s)?(?:[\\s{\\(]\\s*' . $mediaTypesExpression . ')\\s*[^{]*+{.*}\\s*}\\s*#misU',
-            function ($matches) use (&$media) {
-                $media .= $matches[0];
-            },
-            $cssWithoutComments
+        $cssSplitForAllowedMediaTypes = preg_split(
+            '#(@media\\s+(?:only\\s)?(?:[\\s{\\(]\\s*' . $mediaTypesExpression . ')\\s*[^{]*+{.*}\\s*}\\s*)#misU',
+            $cssWithoutComments,
+            -1,
+            PREG_SPLIT_DELIM_CAPTURE
         );
 
-        // filter the CSS
-        $search = [
+        // filter the CSS outside/between allowed @media rules
+        $cssCleaningMatchers = [
             'import/charset directives' => '/^\\s*@(?:import|charset)\\s[^;]+;/misU',
             'remaining media enclosures' => '/^\\s*@media\\s[^{]+{(.*)}\\s*}\\s/misU',
         ];
 
-        $cleanedCss = preg_replace($search, '', $cssForAllowedMediaTypes);
+        $splitCss = [];
+        foreach ($cssSplitForAllowedMediaTypes as $splitPartIndexWhereOddIsMediaRule => $cssPart) {
+            if ($splitPartIndexWhereOddIsMediaRule % 2 !== 0) {
+                // @media rule
+                preg_match('/^([^{]*+){(.*)}[^}]*+$/s', $cssPart, $matches);
+                $media = $matches[1];
+                $css = $matches[2];
+            } else {
+                // CSS outside/between allowed @media rules
+                $media = '';
+                $css = preg_replace($cssCleaningMatchers, '', $cssPart);
+            }
+            $splitCss[] = compact('css', 'media');
+        }
+        return $splitCss;
+    }
 
-        return ['css' => $cleanedCss, 'media' => $media];
+    /**
+     * Temporary method pending refactoring of `parseCssRules` and `copyCssWithMediaToStyleNode`.
+     * Combines the parts returned by `splitCssAndMediaQuery` into just two strings.
+     *
+     * @param string[][] As returned by `splitCssAndMediaQuery`
+     *
+     * @return string[] Array with the following keys:
+     *                  "css" => the cleaned CSS without any @media rules
+     *                  "media" => the valuable @media rules from the original CSS
+     */
+    private function combineSplitCssAndMediaQueryParts($splitCss)
+    {
+        return [
+            'css' => implode('', array_map(
+                function ($splitCssPart) {
+                    if ($splitCssPart['media'] === '') {
+                        return $splitCssPart['css'];
+                    } else {
+                        return '';
+                    }
+                },
+                $splitCss
+            )),
+            'media' => implode('', array_map(
+                function ($splitCssPart) {
+                    if ($splitCssPart['media'] !== '') {
+                        return $splitCssPart['media'] . '{' . $splitCssPart['css'] . '}';
+                    } else {
+                        return '';
```

Comment:
Shorter:

```php
return $splitCssPart['media'] !== '' ? ($splitCssPart['media'] . '{' . $splitCssPart['css'] . '}') : '';
```

Reply:
Changed; however as this would result in exceeding the 120 character line length limit, and CodeSniffer does not allow ternary conditionals to be split onto multiple lines, I have added `extract` to assign the array values to local variables and allow the ternary conditional to be shorter.